text/0093-metric-instruments.md

```diff
@@ -0,0 +1,104 @@
+# List of new metric instruments
+
+Formalize the new metric instruments proposed in [OTEP 88](https://github.com/open-telemetry/oteps/pull/88).
+
+## Motivation
+
+OTEP 88 introduced a framework for reasoning about new metric
+instruments with various refinements and ended with a [sample
+proposal](https://github.com/open-telemetry/oteps/pull/88#sample-proposal).
+This proposal uses that proposal as a starting point.
+
+## Explanation
+
+The four instrument refinements discussed in OTEP 88 are:
+
+* Sum-only: When computing only a sum is the instrument's primary purpose
+* Non-negative: When negative values are invalid
+* Precomputed-sum: When the application has computed a cumulative sum itself
+* Non-negative-rate: When a negative rate is invalid.
+
+OTEP 88 describes how we are meant to compute rate information from
+instruments having the Non-negative-rate refinement.  Temporal
+aggregation (over time) must be treated as a special case, compared
+with spatial aggregation (over labels).  The logic for computing rates
+depends on whether the Precomputed-sum refinement is present or not,
+which determines whether Delta or Cumulative values are being
+captured.
+
+OTEP 88 proposes that when adding new instruments, we specify
+instruments having a single purpose, each with a distinct set of
+refinements, and each with a carefully selected name for the
+properties of the instrument.
+
+OTEP 88 also also proposes to support language-specific specialization
+as well, to support built-in value types (e.g., timestamps).
+
+## Internal details
+
+The existing specification includes three instruments.  In this
+proposal, the two foundational instruments become abstract, in the
+sense that all synchronous instruments are instances of a Measure
+instrument and all asynchronous instruments are instances of an
+Observer instrument.  New instrument names are given to the
+unrestricted, unrefined versions of rthe foundational instruments:
+
+1. **Distribution** is an an unrefined Measure instrument.  Distribution accepts positive or negative values and uses MinMaxSumCount aggregation by default.
```

Comment:
As a naming nit: if the term `Measure` doesn't appear in the user-facing API I think it would be simpler to call this `Measure` and name the abstract type (which AFAICT is only used internally) to something like `AbstractMeasure`.

Reply:
The abstract Measure doesn't need a name, it's the concrete instrument without refinements we need to name. There was debate over this and I don't feel extremely good about either option right now. I like "Counter" because it sounds like a real kind of Instrument you'd use for counting--it's not a figurative word. "Observer" is a thing that observes, I'm happy with it too. 

"Gauge" was figurative, like "reading a gauge", except that the operation is to set something.  "Gauger" doesn't make sense, I feel it was not a good instrument name.

By these criteria, neither Distribution nor Measure is not a good name, they describe the thing the instrument produces, not the thing the instrument does. Perhaps `Recorder` is makes more sense?

This is at the very least making me want to remove "Measure" and "Observer" and refer to synchronous and asynchronous instruments everywhere.

Reply:
This topic is addressed in #96.

Reply:
I've added a note that the names in this document are not finalized.  See #96.